libs/test/Test/Golden.idr

```diff
@@ -89,27 +91,49 @@ record Options where
   interactive  : Bool
   ||| Should we time and display the tests
   timing       : Bool
+  ||| How many threads should we use?
+  threads      : Nat
+
+export
+initOptions : String -> Options
+initOptions exe
+  = MkOptions exe
+              Nothing
+              []
+              False
+              False
+              1
 
 export
 usage : String -> String
-usage exe = unwords ["Usage:", exe, "runtests <path> [--timing] [--interactive] [--cg CODEGEN] [--only [NAMES]]"]
+usage exe = unwords
+  ["Usage:", exe
+  , "runtests <path>"
+  , "[--timing]"
+  , "[--interactive]"
+  , "[--cg CODEGEN]"
+  , "[--threads N]"
+  , "[--only [NAMES]]"
+  ]
 
 ||| Process the command line options.
 export
 options : List String -> Maybe Options
 options args = case args of
-    (_ :: exeUnderTest :: rest) => go rest (MkOptions exeUnderTest Nothing [] False False)
+    (_ :: exe :: rest) => go rest (initOptions exe)
     _ => Nothing
 
   where
 
     go : List String -> Options -> Maybe Options
     go rest opts = case rest of
-      []                      => pure opts
-      ("--timing" :: xs)      => go xs (record { timing = True} opts)
-      ("--interactive" :: xs) => go xs (record { interactive = True } opts)
-      ("--cg" :: cg :: xs)    => go xs (record { codegen = Just cg } opts)
-      ("--only" :: xs)        => pure $ record { onlyNames = xs } opts
+      []                       => pure opts
+      ("--timing" :: xs)       => go xs (record { timing = True} opts)
```

Comment:
Maybe, a space after `True` just for aesthetics?

Reply:
Next time around maybe. :D